src/com/lilithsthrone/main/Main.java

```diff
@@ -871,6 +862,24 @@ public static void saveGame(String name, boolean allowOverwrite) {
 		}
 	}
 
+	public static boolean checkFileName(String name) {
+		if (name.length()==0) {
+			Main.game.flashMessage(PresetColour.GENERIC_BAD, "Name too short!");
+			return true;
+		}
+		if (name.length() > 64) {
+			Main.game.flashMessage(PresetColour.GENERIC_BAD, "Name too long!");
+			return true;
+		}
+		for(String c : "<>:\"/\\|?*".split("")) {
+			if(name.contains(c)) {
+				Main.game.flashMessage(PresetColour.GENERIC_BAD, "Incompatible characters!");
+				return true;
+			}
+		}
+		return false;
+	}
```

Comment:
It is generally considered safer to have a list of allowed characters, instead of disallowed characters.

Also, maybe we should attempt to transform the name, instead of rejecting it, for example: a character name `Ra:t'hou Fh*a` might become the save name `Ra-t-hou_Fh-a`.... i.e. spaces become `_` and everything else becomes `-` so that the final allowed characters are `[A-Za-z0-9_-]`.

This is possible because the name conversion doesn't need to be reversible, only repeatable.

Reply:
The characters I'm rejecting are the danger characters for Windows, Mac and Linux (mostly windows)
From the comments in code I think it used to be an alphanumeric whitelist but was converted to a blacklist, I've just expanded on that